### PR TITLE
Add context doc

### PR DESCRIPTION
**Pull request by justinfagnani**
This adds a *Components > Context* page.

I'm not sure that's the best place in the long term, but for the next single page in component concepts it makes sense. I think I'd rather have a *Managing Data* section that also includes state managers, some conceptual information about difference ways to get state, maybe Task, etc. We could try to add that section now with just *Context* in it.

### CODE REVIEW

File: packages/lit-dev-content/site/docs/components/context.md
Comment (graynorton) [inline packages/lit-dev-content/site/docs/components/context.md:22]: very *similar

File: packages/lit-dev-content/site/docs/components/context.md
Comment (graynorton) [inline packages/lit-dev-content/site/docs/components/context.md:61]: Seems like this should be importing from `'./logger-context.js`

File: packages/lit-dev-content/site/docs/components/context.md
Comment (kevinpschaaf) [inline packages/lit-dev-content/site/docs/components/context.md:270]: I know what you mean here, but it's hard to phrase in an understandable way.

Maybe
```suggestion
Context can be used to pass data from a parent to its light DOM children. Since the parent does not create the children, it cannot leverage template-based data-binding to pass data to its children, but can listen to and respond to `context-request` events.
```

File: packages/lit-dev-content/site/docs/components/context.md
Comment (kevinpschaaf) [inline packages/lit-dev-content/site/docs/components/context.md:281]: I don't super love this example; it's definitely valid, but could we choose one where both are visual (rather than the headless/utility element example)?

File: packages/lit-dev-content/site/docs/components/context.md
Comment (kevinpschaaf) [inline packages/lit-dev-content/site/docs/components/context.md:285]: 😉

**Review by kevinpschaaf — APPROVED** [review]
Looks great. A couple nits and one-or-two stronger suggestions. Otherwise LGTM.

**Comment by justinfagnani** [comment]
@kevinpschaaf thanks for the review!

This PR puts the doc at `/docs/components/context/`. I think there's still a big open question of whether it should go there or somewhere like `/docs/managing-data/context/` as I sketched out in https://github.com/lit/lit.dev/pull/1038

I'll address all the comments here, but not submit yet until we all decide on the final location. For me, the topics of Context, Task, state management integrations, and top-down data flow concepts, push towards a separate data section, fwiw.

File: packages/lit-dev-content/site/docs/components/context.md
Comment (justinfagnani) [reply packages/lit-dev-content/site/docs/components/context.md:252]: thanks!

File: packages/lit-dev-content/site/docs/components/context.md
Comment (justinfagnani) [reply packages/lit-dev-content/site/docs/components/context.md:285]: it's a great example!

File: packages/lit-dev-content/site/docs/components/context.md
Comment (justinfagnani) [reply packages/lit-dev-content/site/docs/components/context.md]: I reworked this a bit. Hopefully it's a bit more clear.

File: packages/lit-dev-content/site/docs/components/context.md
Comment (justinfagnani) [reply packages/lit-dev-content/site/docs/components/context.md]: I think because we're not showing anything visual, that this might be easier to understand.

It is a case a came across a few times actually, where an editor maintainer said they didn't see the value in custom elements because you had to configure the editor in JS anyway.

File: packages/lit-dev-content/site/docs/components/context.md
Comment (arthurevans) [inline packages/lit-dev-content/site/docs/components/context.md:140]:
```suggestion
If you want two _separate_ `createContext()` calls to refer to the same
```

Comment (arthurevans) [reply packages/lit-dev-content/site/docs/components/context.md]: 👋 
Just dropping in to say hi and correct a typo for old time's sake 😂 ... Apparently I haven't turned off my GitHub notifications yet ...

**Comment by arthurevans** [comment]
FWIW I like the idea of a separate data section. Given the other section titles are terse, mostly a single noun, "DATA" or "DATA FLOW" might fit better. I think putting the important word first makes it easier to scan (and we don't have any other section titles with gerunds).

**Comment by justinfagnani** [comment]
I think we're going with a new section, as in https://github.com/lit/lit.dev/pull/1038

I'm not so sure about the title... I'm a little worried that just "Data" isn't descriptive enough to be inviting. "Data Flow" is a bit specific if we also talk about task and state management in there.

So I kept the "Managing Data" section label for now, but made the URL just `/docs/data/` so that we have some flexibility with the label going forward. If we wanted to change the URL too, we could with a redirect.

**Pull request merged**
